clusterDeployer.py

```diff
@@ -221,6 +221,25 @@ def teardown(self) -> None:
 
         print(lh.run(f"ip link set eno1 nomaster"))
 
+    def _validate_external_port(self, lh):
+        # external interface only needed with BM or BF hosts
+        # validation is succesful, but the name is irrelevant
+        if all(x["type"] == "vm" for x in self._cc["workers"]):
+            return True
+
+        # do automatic detection, if needed
+        if self._cc["exteranl_port"] == "auto":
+            self._cc["external_port"] = common.extract_port(lh.run("ip -json r").out, "default")
+            if not self._cc["external_port"]:
+                 return False
+
+        # check that the interface really exists
+        extif = self._cc["external_port"]
+        if lh.run(f"ip link show {extif}").returncode == 0:
```

Comment:
could be something like 

```
if extif not in lh.ports()
``` 

where ports() gets you the list of ports while it parses the json behind the scenes

Reply:
I think Host.if_exists(interface_name) could be useful (and should be faster FWIW)